code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DefaultDeploymentUnitInstance.java

```diff
@@ -330,7 +331,19 @@ protected ServiceIndex createServiceIndex() {
                     InstanceConstants.FIELD_REQUESTED_IP_ADDRESS);
             String requestedIp = null;
             if (requestedIpObj != null) {
-                requestedIp = requestedIpObj.toString();
+                List<String> availableIPs = new ArrayList<>();
+                if (requestedIpObj instanceof List<?>) {
+                    for (Object item : (List<?>) requestedIpObj) {
+                        availableIPs.add(ObjectUtils.toString(item, null));
+                    }
+                }
+                List<String> usedIPs = context.serviceDao.getUsedIPsFromServiceIndexByServiceID(service);
```

Comment:
@alena1108 My original thoughts: At createServiceIndex(), we only select available IpAddress from `requestedIpAddress` fields, which is not used by any active service index. Once ip is selected, `context.sdService.allocateIpToServiceIndex` should validate whether ip is used by other services. Before my changes, this part of code also doesn't do any validation against select IP, it just basically get the IP from `requestedIPAddress` and pass it down. But since we are making this field array of string we need to make sure that inside the same service, the ip is not allocated by another service index.